### ISSUE TITLE
FILTER inside ELSE inside shorthand CASE fails

### ISSUE THREAD

**Issue by splilly** [neo4j/neo4j#9685]
When using a FILTER inside an ELSE of a CASE that is written using the shorthand version of CASE the Cypher will fail. Rewriting to either not use a ELSE or not use the shorthand version of CASE works.

**Neo4j Version:** 3.2.2  
**Operating System:** Windows 10, maxOS, and Ubuntu 16.04
**API:** Cypher

### Steps to reproduce
1. Execute the Cypher 
```
MERGE (:test)<-[:rel]-(:test_rel{prop:true})
MERGE (:test)<-[:rel]-(:test_rel{prop:false})

WITH COUNT(*) AS JUNK

MATCH (t:test)
WITH DISTINCT t AS t
WITH COLLECT(t) AS ts
WITH
    FILTER(t IN ts WHERE (t)<-[:rel]-(:test_rel)) AS all, 3 AS scenario
WITH 
CASE scenario
    WHEN 0 THEN []
    ELSE FILTER(t IN all WHERE (t)<-[:rel]-(:test_rel{prop:true})) 
END AS some

RETURN COUNT(some)
```
2. Execute the Cypher
```
MERGE (:test)<-[:rel]-(:test_rel{prop:true})
MERGE (:test)<-[:rel]-(:test_rel{prop:false})

WITH COUNT(*) AS JUNK

MATCH (t:test)
WITH DISTINCT t AS t
WITH COLLECT(t) AS ts
WITH
    FILTER(t IN ts WHERE (t)<-[:rel]-(:test_rel)) AS all, 3 AS scenario
WITH 
CASE
    WHEN scenario = 0 THEN []
    ELSE FILTER(t IN all WHERE (t)<-[:rel]-(:test_rel{prop:true})) 
END AS some

RETURN COUNT(some)
```
3. Execute the Cypher
```
MERGE (:test)<-[:rel]-(:test_rel{prop:true})
MERGE (:test)<-[:rel]-(:test_rel{prop:false})

WITH COUNT(*) AS JUNK

MATCH (t:test)
WITH DISTINCT t AS t
WITH COLLECT(t) AS ts
WITH
    FILTER(t IN ts WHERE (t)<-[:rel]-(:test_rel)) AS all, 3 AS scenario
WITH 
CASE scenario
    WHEN 0 THEN []
    WHEN 3 THEN FILTER(t IN all WHERE (t)<-[:rel]-(:test_rel{prop:true})) 
END AS some

RETURN COUNT(some)
```

### Expected behavior
All three steps should return the same result.

### Actual behavior
Step 1 fails but steps 2 and 3 return the correct result.

**Comment by sherfert** [comment]
Thanks for reporting, we will pick that up.

**Comment by sherfert** [comment]
This fix will be part of 3.2.7
